Assets/Scripts/OrbitParent.cs

```diff
@@ -22,7 +25,11 @@ void Update()
         }
         if (isRotating)
         {
-            transform.RotateAround(circle.position, zAxis, speed * Time.deltaTime);
+            rotation_amount = (rotation_amount + speed * Time.deltaTime) % 360f;
         }
+
+        transform.position = circle.position + Vector3.down;
```

Comment:
can you add a comment explaining that you reset the position please?

Reply:
Done